### ISSUE TITLE
Styles are not overwritten 

### ISSUE THREAD

**Issue by ivanbanov** [gregberge/xstyled#287]
Im not really sure if it's intended or an actual bug.

The styles props are not being overwriten when extending the "default box" `x.div`

![image](https://user-images.githubusercontent.com/2953018/123647045-0b037700-d828-11eb-9104-4c1511663a1c.png)

A repo with a quick reproduction can be found here https://github.com/ivanbanov/xstyled-styles-override-bug/

**Comment by ivanbanov** [comment]
ok, nevermind I will close this one since adding the `importMap` fixes the problem

**Comment by ivanbanov** [comment]
actually the first css prop was not transpiled when using the plugin 
![image](https://user-images.githubusercontent.com/2953018/123666485-8a4d7680-d839-11eb-9cf6-b0571f5948ce.png)

**Comment by agriffis** [comment]
@ivanbanov You are reporting two things here:

1. Styles aren't overriding the way you expect
2. Theme values aren't being replaced

The second issue is #285 so let's not repeat it here.

For the first issue, the problem is that you're wrapping a system component (that receives props) with a styled component (that sets styles in a style object). The wrapper wins, and this is correct! It would be unexpected for an inner component's styling to override a wrapper component.

To make it work the way you want, pass `system` to the styled wrapper as an additional style generator:

```
const InfoBox = styled(Box)({ backgroundColor: 'info' }, system)
```

At this point, you no longer need `Box`, you might as well use `div`:

```
const InfoBox = styled.div({ backgroundColor: 'info' }, system)
```

and there is also a shortcut for this, which automatically applies `system`:

```
const InfoBox = styled.box({ backgroundColor: 'info' })
```

Alternatively you can use `<x.div>` which works the same as `styled.box`:

```
const InfoBox = props => <x.div backgroundColor='info' {...props} />
```

I made a fork to demonstrate each of these approaches, see https://github.com/agriffis/xstyled-styles-override-bug/blob/master/index.js#L5-L9

**Comment by ivanbanov** [comment]
Cool thank, I ended up using a utility which is also working with the `importMap` settings

```jsx
import styled from '@xstyled/emotion'
import { system } from '@xstyled/system'

export const box = (...styles) => styled.div(...styles, system)

// usage

const StyledComponent = box({ color: 'info' })
<StyledComponent color='error' />
```

Again, I appreciate your support :)
